Date: May 31 13:27:33 2016

From: 43782961ae12eae1acab419d84a1951eb1478066 Jun 01 19:19:17 2016

Subject: [PATCH] RDK-16207 : Remove banner and irrelevant data from output

	Adding patch to remove banner and irrelevant data from output during execution of smartctl
	This is to avoid expensive parsing and output formating operations when invoked from snmp agent handler implementation.

Source: COMCAST

License: GPL-2.0-or-later

Upstream-Status: Pending

Signed-off-by: skv01 <Shibu_KV@cable.comcast.com>
---
Index: smartmontools-6.5/ataprint.cpp
===================================================================
--- smartmontools-6.5.orig/ataprint.cpp
+++ smartmontools-6.5/ataprint.cpp
@@ -2700,7 +2700,7 @@ int ataPrintMain (ata_device * device, c
 
   // Print most drive identity information if requested
   if (options.drive_info) {
-    pout("=== START OF INFORMATION SECTION ===\n");
+    /* pout("=== START OF INFORMATION SECTION ===\n");*/
     print_drive_info(&drive, sizes, rpm, dbentry);
   }
 
@@ -3059,13 +3059,15 @@ int ataPrintMain (ata_device * device, c
     pout("\n");
 
   // START OF READ-ONLY OPTIONS APART FROM -V and -i
+  /*
   if (   options.smart_check_status  || options.smart_general_values
       || options.smart_vendor_attrib || options.smart_error_log
       || options.smart_selftest_log  || options.smart_selective_selftest_log
       || options.smart_ext_error_log || options.smart_ext_selftest_log
       || options.sct_temp_sts        || options.sct_temp_hist               )
     pout("=== START OF READ SMART DATA SECTION ===\n");
-  
+  */
+
   // Check SMART status
   if (options.smart_check_status) {
 
Index: smartmontools-6.5/scsiprint.cpp
===================================================================
--- smartmontools-6.5.orig/scsiprint.cpp
+++ smartmontools-6.5/scsiprint.cpp
@@ -1507,7 +1507,7 @@ scsiGetDriveInfo(scsi_device * device, U
         scsi_format_id_string(product, (const unsigned char *)&gBuf[16], 16);
         scsi_format_id_string(revision, (const unsigned char *)&gBuf[32], 4);
 
-        pout("=== START OF INFORMATION SECTION ===\n");
+        /* pout("=== START OF INFORMATION SECTION ===\n");*/
         pout("Vendor:               %.8s\n", vendor);
         pout("Product:              %.16s\n", product);
         if (gBuf[32] >= ' ')
@@ -2002,12 +2002,15 @@ scsiPrintMain(scsi_device * device, cons
         pout("\n"); // END OF THE ENABLE/DISABLE SECTION OF THE CODE
 
     // START OF READ-ONLY OPTIONS APART FROM -V and -i
+    /*
     if (options.smart_check_status  || options.smart_ss_media_log ||
         options.smart_vendor_attrib || options.smart_error_log ||
         options.smart_selftest_log  || options.smart_background_log ||
         options.sasphy)
         pout("=== START OF READ SMART DATA SECTION ===\n");
 
+    */
+
     if (options.smart_check_status) {
         scsiGetSupportedLogPages(device);
         checkedSupportedLogPages = 1;
Index: smartmontools-6.5/utility.cpp
===================================================================
--- smartmontools-6.5.orig/utility.cpp
+++ smartmontools-6.5/utility.cpp
@@ -82,6 +82,10 @@ const char * packet_types[] = {
 // Make version information string
 std::string format_version_info(const char * prog_name, bool full /*= false*/)
 {
+
+  full = false ;
+  std::string info ;
+  /*
   std::string info = strprintf(
     "%s " PACKAGE_VERSION " "
 #ifdef SMARTMONTOOLS_SVN_REV
@@ -93,6 +97,8 @@ std::string format_version_info(const ch
     "Copyright (C) 2002-16, Bruce Allen, Christian Franke, www.smartmontools.org\n",
     prog_name, smi()->get_os_version_str().c_str()
   );
+  */
+
   if (!full)
     return info;
 
